server/src/main/java/jake2/server/SV_MAIN.java

```diff
@@ -904,23 +904,24 @@ void SV_GiveMsec() {
      */
     private void SV_FinalMessage(String message, boolean reconnect) {
 
-        Globals.net_message.clear();
-        new PrintMessage(Defines.PRINT_HIGH, message).writeTo(Globals.net_message);
+        sizebuf_t buffer = new sizebuf_t();
+        buffer.init(new byte[MAX_MSGLEN], MAX_MSGLEN);
+        new PrintMessage(Defines.PRINT_HIGH, message).writeTo(buffer);
 
         if (reconnect)
-            new ReconnectMessage().writeTo(Globals.net_message);
+            new ReconnectMessage().writeTo(buffer);
         else
-            new DisconnectMessage().writeTo(Globals.net_message);
```

Comment:
does "write_to" completely override the buffer or only appends to it? 
I wonder if there are some subtle "bugs" where some messages are used in a `writeTo` but never  in `Transmit`

Reply:
It appends to the buffer. The same contract as with the standard ByteBuffers and OutputStreams.
Transmit happens on every frame so it should sent whatever has accumulated during the frame update